MovieApp/MovieUseCase/MovieUseCase.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+
+class MovieUseCase {
+    
+    func getPopularMovies(criteria: String) async -> [MovieListModel] {
+        guard let request = urlRequest(for: "https://five-ios-api.herokuapp.com/api/v1/movie/popular?criteria=\(criteria)")
+        else { return [] }
+        
+        async let (data, _) = try await URLSession.shared.data(for: request)
```

Comment:
This works, but it possibly creates several problems... We usually do not declare variables as async unless there are several of them. I know why you did it: probably to avoid declaring that this function throws an error....
Here is a better solution:
```
        let response = try? await URLSession.shared.data(for: request)
        guard
            let data = response?.0,
            let movies = try? JSONDecoder().decode([MovieListModel].self, from: data)
        else { return [] }
        return movies
```